feat: add support for macOS

chungdaniel opened the pull request:
Primary changes:
- Move iOS code `sharedDarwinSource` to reuse for macOS according to https://docs.flutter.dev/packages-and-plugins/developing-packages#shared-ios-and-macos-implementations - thanks to @jointhejourney for their work in #156 for the idea
- Added support for macOS in example app

Extra:
- Added swift_version to `amplitude_flutter.podspec`, follow-up to #211 for v4
- Bumped flutter depdendency to 3.7.0+ to support `sharedDarwinSource`

Note:
- PR should probably be focused mostly on non-example files

Addresses #46 
Jira: [[SDK - Flutter] Flutter macOS Support](https://amplitude.atlassian.net/browse/AMP-35292)

chungdaniel committed:
Move iOS files to darwin and symlink

chungdaniel committed:
feat: support macOS for flutter SDK
- use symlinks to share code between iOS and macOS, could not use sharedDarwinResource flag

chungdaniel committed:
add gitignore files

chungdaniel committed:
add macos example

chungdaniel committed:
add podfile and runner.csxcheme

chungdaniel committed:
Try using sharedDarwinSource

chungdaniel committed:
Change flutter min version

chungdaniel committed:
Update swift-lint.yml to enter `darwin` instead of `ios`

Mercy811 commented on README.md:
```diff
@@ -22,9 +22,9 @@ From Amplitude Flutter v4, we bump up the kotlin version to v1.9.22 to support l
+|  Dart   | Flutter | Amplitude Flutter | Gradle | Android Gradle Plugin | Kotlin Gradle Plugin |
+|---------|---------|-------------------|-------|-----------------------|-----------------------|
+| `3.7.0+`|`3.3+`   | `4.+`             | `8.2` | `8.2.2`               | `1.9.22`              |
 
```
What requires Dart version? `sharedDarwinSource` requires Flutter 3.7 right?

chungdaniel replied:
Ah, good catch. Yes, Flutter requires 3.7 for `sharedDarwinSource`, but we need dart 3.3+ to support Flutter web after adopting `js_interop`. I'll swap those around

izaaz commented on darwin/amplitude_flutter.podspec:
```diff
@@ -0,0 +1,33 @@
+
+  s.pod_target_xcconfig = { 'DEFINES_MODULE' => 'YES' }
+  s.swift_version = '5.9'
+  s.dependency 'AmplitudeSwift', '~> 1.6'
```
Can we update to the latest version of the SDK as the min version?

izaaz commented on darwin/Resources/PrivacyInfo.xcprivacy:
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Do we need to add this? The underlying SDK also has a privacy manifest which is more up to date.

chungdaniel replied:
It was added by default when I tried adding macOS as a platform as a result of me updating to latest flutter locally. I think it should be fine to remove it.

izaaz reviewed (approved):
Mostly LGTM with a few minor comments.

Mercy811 reviewed (approved):
Thank you @chungdaniel, LGTM other than flutter & dart version requirements.

chungdaniel committed:
address PR comments:
- swap dart and flutter versions correctly
- use latest version of amplitude swift
- remove privacyinfo.xcprivacy
